model/src/main/java/io/syndesis/model/connection/Action.java

```diff
@@ -54,4 +54,6 @@ default Action withId(String id) {
     class Builder extends ImmutableAction.Builder {
     }
 
+    ActionMetadata metadata();
```

Comment:
Could we call it something different like `metadata` ? Maybe `definition` ?

The reason why I ask is, that there is still the idea that we change our resource model to a more openshift/kubernetes like format, where we mimic the k8s resource format here. The reasoning for this is, that this keeps the door somewhat open, that in a future version we could easily migrate to K8s _customer resource definitions_ (CRD) which e.g. use secrets etc. directly. 

It's not that far, but `metadata` is a _reserved_ word for such a resoruce definition which all look like:

```yaml
apiVersion:
kind:
metadata:
   ...
spec:
   ...
status:
  ....
```

Also I think that _definition_ is a bit more specific than _metadata_